docs/basic-serialization.md

```diff
@@ -0,0 +1,656 @@
+<!--- TEST_NAME BasicSerializationTest -->
+
+# Basic Serialization
+
+This is the first chapter of the [Kotlin Serialization Guide](serialization-guide.md).
+This chapter shows the basic use of Kotlin Serialization and explains its core concepts.
+
+**Table of contents**
+
+<!--- TOC -->
+
+* [Basics](#basics)
+  * [JSON encoding](#json-encoding)
+  * [JSON decoding](#json-decoding)
+* [Serializable classes](#serializable-classes)
```

Comment:
I'd suggest to restructure this document a bit. It is perfectly suited for thorough reading, but sometimes people just want to know 'how do i rename a property in json' or 'how to make it optional' and want to look that up quick. They also may not know that such functionality is called 'serial field names' or something.

To me, it makes sense to reorder contents in a such way that most commonly used functionality is placed first -- i.e. serial names, optionals and transients should be in the top. Given that, most of the people would be able to spend 2-3 minutes to grasp first half of the document or just necessary chapters, and already use serialization framework efficiently